ui/src/apis/beaconchaApi/index.ts

```diff
@@ -1,37 +1,29 @@
 import { BeaconchaGetResponse } from "./types";
-import { ApiParams } from "../../types";
 import { Web3signerGetResponse } from "../web3signerApi/types";
+import { StandardApi } from "../standardApi";
+import { maxValidatorsPerRequest } from "../../params";
 
-//TODO Use inheritance for both APIs?
-export class BeaconchaApi {
-  baseUrl: string;
-  endpoint: string;
-  fullUrl: string;
-
-  constructor(apiParams: ApiParams) {
-    this.endpoint = apiParams.apiPath;
-    this.baseUrl = apiParams.baseUrl;
-    this.fullUrl = this.baseUrl + this.endpoint;
-  }
-
+export class BeaconchaApi extends StandardApi {
   /*
    * Fetch info for every validator PK
    */
   public async fetchAllValidatorsInfo({
-    beaconchaApi,
     keystoresGet,
   }: {
-    beaconchaApi: BeaconchaApi;
     keystoresGet: Web3signerGetResponse;
   }): Promise<BeaconchaGetResponse[]> {
     const validatorsInfo = new Array<BeaconchaGetResponse>();
+
     let allValidatorPKs = keystoresGet.data.map(
       (keystoreData) => keystoreData.validating_pubkey
     );
 
-    for await (const validatorPK of allValidatorPKs) {
-      const validatorInfo = await beaconchaApi.fetchValidatorInfo(validatorPK);
-      validatorsInfo.push(validatorInfo);
+    const chunkSize = maxValidatorsPerRequest;
+
+    for (let i = 0; i < allValidatorPKs.length; i += chunkSize) {
```

Comment:
I do not think there is a much better way, because each GET request must include up to 100 validator pubkeys, so the total array of PKs must be split into 100-PKs chunks so that a GET request can be performed for each of them.